### PR TITLE
Add architecture doc on globalization invariant mode

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

**Comment by tarekgh** [comment]
LGTM

File: Documentation/architecture/globalization-invariant-mode.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+Globalization support has the following potential challenges for applications:
+
+* Different behavior across OSes (and potentially OS versions).
+* Installing/carrying the [ICU](http://icu-project.org) package on Linux (~28 MB).
+
+Note: On Linux, .NET Core relies on globalization data from ICU. For example, [.NET Core Linux Docker images](https://github.com/dotnet/dotnet-docker/blob/master/2.0/runtime-deps/stretch/Dockerfile) intall this component. Globalization data is available on Windows and macOS as part of their base installs.
+  
+## Cultures and culture data
+ 
+When enabling the invariant mode, all cultures behave like the invariant culture. The invariant culture hs the following characteristics:
```
Comment (stephentoub) [inline Documentation/architecture/globalization-invariant-mode.md]: Nit: hs => has

File: Documentation/architecture/globalization-invariant-mode.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+Note: On Linux, .NET Core relies on globalization data from ICU. For example, [.NET Core Linux Docker images](https://github.com/dotnet/dotnet-docker/blob/master/2.0/runtime-deps/stretch/Dockerfile) intall this component. Globalization data is available on Windows and macOS as part of their base installs.
+  
+## Cultures and culture data
+ 
+When enabling the invariant mode, all cultures behave like the invariant culture. The invariant culture hs the following characteristics: 
+ 
+* Culture names (English, native display, ISO, language names) will return invariant names. For instance, when requesting culture native name, you will get "Invariant Language (Invariant Country)".
+* All cultures LCID will have value 0x1000 (which means Custom Locale ID). The exception is the invariant cultures which will still have 0x7F.
+* All culture parents will be invariant. In other word, there will not be any neutral cultures by default but the apps can still create a culture like "en"
```
Comment (stephentoub) [inline Documentation/architecture/globalization-invariant-mode.md]: Nit: missing period at end of sentence.

File: Documentation/architecture/globalization-invariant-mode.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+* The application can still create any culture (e.g. "en-US") but all the culture data will still be driven from the Invariant culture. Also, the culture name used to create the culture should conform to [BCP 47 specs](https://tools.ietf.org/html/bcp47).
+* All Date/Time formatting and parsing will use fixed date and time patterns. For example, the short date will be "MM/dd/yyyy" regardless of the culture used. Applications having old formatted date/time strings may not be able to parse such string without using ParseExact.
```
Comment (stephentoub) [inline Documentation/architecture/globalization-invariant-mode.md]: Nit: parse such string => parse such strings

File: Documentation/architecture/globalization-invariant-mode.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+* Numbers will always be formatted as the invariant culture. For example, decimal point will always be formatted as ".". Number strings previously formatted with cultures that have different symbols will fail parsing.
+* All cultures will have currency symbol as "¤"
+* Culture enumeration will always return a list with one culture which is the invariant culture.
+ 
+## String casing
+ 
+String casing (ToUpper and ToLower) will be performed for the ASCII range only. Requests to case code points outside that range will not be performed,however no exception will be thrown. In other words, casing will only be performed for character range ['a'..'z'].
```
Comment (stephentoub) [inline Documentation/architecture/globalization-invariant-mode.md]: Nit: missing space before "however no exceptions.

File: Documentation/architecture/globalization-invariant-mode.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+* Turkish culture, using 
+* CompareOptions.Ignorecase
+
+However, the following comparison will resolve to being equal:
+
+* 'i', compared to
+* 'I', using 
+* CompareOptions.Ignorecase
+ 
+It is worth to notice that all other [sort comparison options](https://docs.microsoft.com/dotnet/api/system.globalization.compareoptions) (for example, ignore symbols, ignore space, Katakana, Hiragana) will have no effect in the invariant mode (they are ignored).
```
Comment (stephentoub) [inline Documentation/architecture/globalization-invariant-mode.md]: Nit: It is worth to notice => It is worth noticing

File: Documentation/architecture/globalization-invariant-mode.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+* If the app requested to normalize any string, the original string is returned without modification. 
+* If the app asked if any string is normalized, the return value will always be `true`.
+ 
+## Internationalized Domain Names (IDN) support
+
+[Internationalized Domain Names](https://en.wikipedia.org/wiki/Internationalized_domain_name) require globalization data to perform conversion to ACII or Unicode forms, which isn't available in the invariant mode. In this mode, IDN functionality has the following behavior:
+
+* IDN support doesn't conform to the latest standard.
+* IDN support will be incorrect if input IDN string is not normalized since normalization is not supported in invariant mode.
```
Comment (stephentoub) [inline Documentation/architecture/globalization-invariant-mode.md]: Nit: if input IDN string => if the input IDN string

File: Documentation/architecture/globalization-invariant-mode.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+* Some basic IDN strings will still produce correct values.
+ 
+## Time zone display name in Linux
+ 
+When running on Linux ICU is used to get the time zone display name. In invariant mode, the standard time zone names are returned instead.
```
Comment (stephentoub) [inline Documentation/architecture/globalization-invariant-mode.md]: Nit: comma after "When running on Linux"

**Comment by richlander** [comment]
Thanks @stephentoub. Applied fixes for your feedback.

**Pull request merged**